File: src/main/java/com/ject/studytrip/stamp/domain/repository/StampCommandRepository.java
package com.ject.studytrip.stamp.domain.repository;

public interface StampCommandRepository {
    boolean existsByTripIdAndCompletedIsFalseAndDeletedAtIsNull(Long tripId);

    long deleteAllByDeletedAtIsNotNull();

    long deleteAllByDeletedTripOwner();

    long deleteAllByMemberId(Long memberId);
}

File: src/main/kotlin/com/ject/studytrip/stamp/domain/repository/StampCommandRepository.kt
package com.ject.studytrip.stamp.domain.repository

interface StampCommandRepository {
    fun existsByTripIdAndCompletedIsFalseAndDeletedAtIsNull(tripId: Long): Boolean

    fun deleteAllByDeletedAtIsNotNull(): Long

    fun deleteAllByDeletedTripOwner(): Long

    fun deleteAllByMemberId(memberId: Long): Long
}
